### PR TITLE
Use CultureInvariant in Regex calls

### PR DESCRIPTION
**Pull request by jennyf19**
Use a static variable
#859

### CODE REVIEW

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Instance/Authenticator.cs [hunk 1]
```diff
@@ -42,7 +42,9 @@ internal class Authenticator
     {
         private const string TenantlessTenantName = "Common";
 
-        private bool updatedFromTemplate;
+        private bool _updatedFromTemplate;
+
+        internal static Regex TenantNameRegex = new Regex(Regex.Escape(TenantlessTenantName), RegexOptions.IgnoreCase | RegexOptions.CultureInvariant);
```
Comment (aadidgit) [inline src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Instance/Authenticator.cs]: CRITICAL Change the visibility of 'TenantNameRegex' or make it 'const' or 'readonly'. [rule](http://iddp-sonarqube:9000/coding_rules#rule_key=csharpsquid%3AS2223)

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Helpers/AdalIdHelper.cs [hunk 2]
```diff
@@ -67,6 +67,8 @@ internal static class AdalIdParameter
     /// </summary>
     internal static class AdalIdHelper
     {
+        internal static Regex AdalVersionRegex = new Regex(@"Version=[\d]+.[\d+]+.[\d]+.[\d]+", RegexOptions.CultureInvariant);
```
Comment (aadidgit) [inline src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Helpers/AdalIdHelper.cs]: CRITICAL Change the visibility of 'AdalVersionRegex' or make it 'const' or 'readonly'. [rule](http://iddp-sonarqube:9000/coding_rules#rule_key=csharpsquid%3AS2223)

**Comment by aadidgit** [comment]
SonarQube analysis reported 2 issues
* ![CRITICAL][CRITICAL] 2 critical

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Instance/Authenticator.cs (hunk 1 above)
Comment (kpanwar) [reply src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Instance/Authenticator.cs]: make it private

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Helpers/AdalIdHelper.cs (hunk 2 above)
Comment (kpanwar) [reply src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Helpers/AdalIdHelper.cs]: make it private

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Instance/Authenticator.cs (hunk 1 above)
Comment (jennyf19) [reply src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Instance/Authenticator.cs]: done.

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Helpers/AdalIdHelper.cs (hunk 2 above)
Comment (jennyf19) [reply src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Helpers/AdalIdHelper.cs]: done.

File: tests/Test.ADAL.NET.Unit/UnitTests.cs [hunk 3]
```diff
@@ -160,6 +160,13 @@ public void AuthenticationParametersTest()
             RunAuthenticationParametersNegative("Bearer authorization_uri=abc=de");
         }
 
+        [TestMethod]
+        [Description("Test for Adal version creation with Regex")]
+        public void AdalVersionRegexTest()
+        {
+            string adalVersion = AdalIdHelper.GetAdalVersion();
+            Assert.AreEqual("0.0.0.0", adalVersion);
```
Comment (brentschmaltz) [inline tests/Test.ADAL.NET.Unit/UnitTests.cs]: curious what the default version is "0.0.0.0".

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Helpers/AdalIdHelper.cs [hunk 4]
```diff
@@ -98,8 +100,7 @@ internal static class AdalIdHelper
         public static string GetAdalVersion()
         {
             string fullVersion =  typeof (AdalIdHelper).GetTypeInfo().Assembly.FullName;
-            Regex regex = new Regex(@"Version=[\d]+.[\d+]+.[\d]+.[\d]+");
-            Match match = regex.Match(fullVersion);
+            Match match = AdalVersionRegex.Match(fullVersion);
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Helpers/AdalIdHelper.cs]: If this is performed a lot, then a faster approach would be to do:
if (AdalVersionRegex.IsMatch(fullVersion)
{
  return fullVersion.Substring("Version".Length);
}
}

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Helpers/AdalIdHelper.cs [hunk 5]
```diff
@@ -67,6 +67,8 @@ internal static class AdalIdParameter
     /// </summary>
     internal static class AdalIdHelper
     {
+        private static readonly Regex AdalVersionRegex = new Regex(@"Version=[\d]+.[\d+]+.[\d]+.[\d]+", RegexOptions.CultureInvariant);
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Helpers/AdalIdHelper.cs]: You may want to think about RegexOptions.Compiled. The first check is slower, but the future are much faster.
https://msdn.microsoft.com/en-us/library/system.text.regularexpressions.regexoptions(v=vs.110).aspx

Comment (jennyf19) [reply src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Helpers/AdalIdHelper.cs]: I like this, but looks like netstandard1.1 doesn't support RegexOptions.Compiled

File: tests/Test.ADAL.NET.Unit/UnitTests.cs (hunk 3 above)
Comment (jennyf19) [reply tests/Test.ADAL.NET.Unit/UnitTests.cs]: This caused a problem w/our tests, so I'm just going to assert that it's not null

**Pull request merged**